Triangulation_2/include/CGAL/Triangulation_2/internal/Polyline_constraint_hierarchy_2.h

```diff
@@ -262,18 +413,42 @@ class Polyline_constraint_hierarchy_2
     return sc_to_c_map.end();
   }
 
+  Edge_iterator edges_begin() const {
+    BOOST_STL_INTERFACES_STATIC_ASSERT_CONCEPT(Edge_iterator, std::bidirectional_iterator);
+    BOOST_STL_INTERFACES_STATIC_ASSERT_ITERATOR_TRAITS(Edge_iterator, std::bidirectional_iterator_tag,
+                                                       std::bidirectional_iterator, Edge, Edge,
+                                                       typename Edge_iterator::pointer, std::ptrdiff_t);
+    return Edge_iterator(Edge_iterator::Construction_access::begin_tag(), &constraint_set);
+  }
+
+  Edge_iterator edges_end() const {
+    return Edge_iterator(Edge_iterator::Construction_access::end_tag(), &constraint_set);
+  }
+
+  auto edges() const { return Iterator_range<Edge_iterator>(edges_begin(), edges_end()); }
+
```

Comment:
@afabri the range `edges()` is a range on `std::pair<Vertex_handle, Vertex_handle>`, one for earch subconstraints.

Suggestion for a better name?

Reply:
As `Edge` is a pair of face handle and an index it should not be `Edge_iterator`but  `Subconstraint_iterator`.